docs/user-guide/reduction-workflow-guidelines.md

```diff
@@ -43,6 +40,56 @@ We plan to include the following in future versions of the guidelines:
+|-------------------------------|-------------|---------------------------------------------------------------------------|
+| --- **Files** ---             |             |                                                                           |
+| Filename \| *Filename         | str         | Simple name of a file, must be processed into FilePath                    |
+| FilePath \| *FilePath         | Path        | Concrete path to a file on the host filesystem, ideally absolute          |
+| --- **Run IDs** ---           |             |                                                                           |
+| SampleRun, BackgroundRun, ... | Any         | Identifier for a run                                                      |
+| RunType                       | TypeVar     | Constrained to the run types used by the package, see above               |
+| RunTitle                      | str         | Extracted from NeXus or provided by user, can be used to find files       |
+| --- **Monitors** ---          |             |                                                                           |
+| IncidentMonitor \| *Monitor   | Any         | Identifier for a monitor                                                  |
+| MonitorType                   | TypeVar     | Constrained to the monitor types used by the package, see above           |
+| --- **Flags** ---             |             |                                                                           |
+| UncertaintyBroadcastMode      | enum        | E.g., `Enum('UncertaintyBroadcastMode', ['drop', 'upper_bound', 'fail'])` |
+| ReturnEvents                  | bool        | Select whether to return events or histograms from the workflow           |
+| CorrectForGravity             | bool        | Toggle gravity correction                                                 |
+| --- **Misc** ---              |             |                                                                           |
+| NeXus*                        | Any         | Spelling of all NeXus-related keys                                        |
+| WavelengthBins \| *Bins       | sc.Variable | Bin-edges                                                                 |
+
+### C.2: Use flexible types
+
+**Reason**
+Users should not have to worry about the concrete type of parameters.
+
+**Example**
+
+- Numbers should use the appropriate abstract type from {mod}`numbers`.
+  E.g.,
+  ```python
+  P = NewType('P', numbers.Real)
+  pipeline[P] = 3.0  # works
+  pipeline[P] = 3    # works, too, but not if `P` were `float`
+  ```
+- Use {class}`collections.abc.Sequence` instead of `list` or `tuple`.
+  - But do *not* use {class}`typing.Iterable`!
+    Parameters may be consumed multiple times and iterables are not guaranteed to support that.
+- Gracefully promote dtypes for small parameters.
+  E.g., `sc.scalar(2, unit='m')` and `sc.scalar(2.0, unit='m')` should be usable interchangeably.
```

Comment:
👍 
Should we require the same, e.g., for params like bin-edges such that, e.g., passing `arange` works just like passing `linspace`?

Reply:
Operations may raise when ints are passed instead of floats as coords (bin edges, for example). I am suggesting extending your suggestion for graceful handling.